src/FakeItEasy/Configuration/ExceptionThrower.StronglyTyped.tt

```diff
@@ -0,0 +1,133 @@
+<#@ template hostspecific="false" language="C#" #>
+<#@ output extension=".cs" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Linq" #>
+<#@ include file="../CodeGen/Ordinals.ttinclude" #>
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a T4 template.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+namespace FakeItEasy.Configuration
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+#if FEATURE_NETCORE_REFLECTION
+    using System.Reflection;
+#endif
+    using FakeItEasy.Core;
+
+    public partial interface IExceptionThrowerConfiguration<out TInterface>
+    {
+<# GenerateMethodDeclarations(); #>
+    }
+
+    internal partial class RuleBuilder
+    {
+        private const string NameOfThrowsFeature = "throws";
+
+<# GenerateMethodImplementations("IVoidConfiguration"); #>
+        public partial class ReturnValueConfiguration<TMember>
+        {
+<# GenerateMethodImplementations("IReturnValueConfiguration<TMember>", "    "); #>
+        }
+    }
+
+    internal partial class AnyCallConfiguration
+    {
+        private const string NameOfThrowsFeature = "throws";
+
+<# GenerateMethodImplementations("IVoidConfiguration"); #>
+    }
+
+    internal partial class PropertySetterConfiguration<TValue>
+    {
+        private const string NameOfThrowsFeature = "throws";
+
+<# GenerateMethodImplementations("IPropertySetterConfiguration"); #>
+        private partial class PropertySetterAdapter
+        {
+<# GenerateMethodImplementations("IPropertySetterConfiguration"); #>
+        }
+    }
+}

```

Comment:
I know we normally put each class in its own file, but it was much easier to do it like this, and anyway it's generated code.